crates/api_models/src/payment_methods.rs

```diff
@@ -144,6 +144,21 @@ pub struct PaymentMethodResponse {
     pub created: Option<time::PrimitiveDateTime>,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, serde::Deserialize, serde::Serialize)]
+pub enum PaymentMethodsData {
+    Card(CardDetailsPaymentMethod),
+    BankDetails,
```

Comment:
If the variant doesn't have data associated with it, lets just not have it for now.